docs/DeveloperGuide.md

```diff
@@ -262,73 +262,227 @@ _{Explain here how the data archiving feature will be implemented}_
+6. HRConnect un-assigns the employee from the project
+
+   Use case ends.
+
+**Extensions**
+
+- 2a. The list is empty.
+
+  Use case ends.
+
+- 4a. The list is empty.
+
+  Use case ends.
+
+- 5a. The given id is invalid.
+
+  - 5a1. HRConnect shows an error message.
+
+    Use case resumes at step 4.
+
+---
+
+**Use case: Filter employees with desired skills**
+
+**MSS**
+
+1. User requests to list skills the employees in the company have
+2. HRConnect shows a list of skills
+3. User selects a few skills
+4. User requests to list all employees with all of the skills selected
+5. HRConnect lists all employees with all of the skills selected
+
+   Use case ends.
+
+**Extensions**
+
+- 2a. The list of skills is empty.
+
+  Use case ends.
+
+- 3a. User selects invalid skills.
+
+  - 3a1. HRConnect shows an error message.
+
+    Use case resumes at step 2.
+
+---
+
+### Non-Functional Requirements
+
+1.  _Technical Requirement_: The App should work on any _mainstream OS_ as long as it has Java `17` or above installed.
+2.  _Performance Requirement_: Should be able to hold up to 1000 employees without a noticeable sluggishness in performance for typical usage.
+3.  _Performance Requirement_: The App should respond to most commands within 200 milliseconds.
+4.  _Quality Requirement_: A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+5.  _Quality Requirement_: A HR professional who has never used software to manage manpower should be able to learn basic operations like adding, deleting and assigning employees to projects within the first 10 minutes of reading the user guide.
+6.  _Fault Tolerance_: The App should be able to handle corrupted data without crashing.
+7.  _Documentation_: The code should be well-documented so that maintainers new to the project can quickly understand and contribute to the codebase.
+8.  _Constraints_: The App should be backward compatible with data produced by earlier versions of the App.
+9.  _Privacy Requirement_: The App should comply with the Personal Data Protection Act (PDPA) in handling personal information.
+10. _Notes about project scope_: The App is not required to handle the actual firing / hiring of an employee or the completion or termination of a project.
+
+### Glossary
+
+- **Mainstream OS**: Windows, Linux, Unix, MacOS
+- **CLI**: Command Line Interface - A way of interacting with a computer system by inputting lines of text. The keyboard is primarily used.
+- **GUI**: Graphical User Interface - A way of interacting with a computer system where graphical elements such as windows, buttons, and menus are used.
+- **Private contact detail**: A contact detail that is not meant to be shared with others
+- **HR**: Human Resource - people responsible for finding, hiring, and training employees
```

Comment:
A bit of a nitpick but it's usually called Human Resources, and its a "department" rather than individual people

Rest look good!

Reply:
Updated the definition of "HR".

Reply:
Nice!